scripts/consumer-scenario-2.ts

```diff
@@ -0,0 +1,85 @@
+// This can be a wallet project
+// This can also be a Verifier-Backend-Server side project
+
+import {EvaluationResults, PEJS, SelectResults, VerifiableCredential} from "@sphereon/pe-js";
+import { PresentationDefinition, PresentationSubmission } from "@sphereon/pe-models";
+import { Wallet } from "./Wallet";
+
+/**
+ * This scenario contains a flow in which Alice wants to prove to Bob that she has some kind of Credentials.
+ * Alice has previously contacted Bob and therefore know what he's expecting.
+ * It means that Bob has already sent her a PresentationDefinition object.
+ * Alice's wallet sends all of her available credentials to PEJS to see which one is useful
+ * In the second example the PresentationDefinition object is a simple one. Bob is sending her a presentationDefinition with two InputDescriptors
+ * Alice has two credential in her wallet that have the properties requested by Bob
+ */
+function checkScenario_2() {
+  const wallet1: { holder: string; verifiableCredentials: VerifiableCredential[] } = new Wallet().getWallet();
+  const wallet2: { holder: string; verifiableCredentials: VerifiableCredential[] } = new Wallet().getWallet();
+  wallet2.verifiableCredentials[2].credentialSubject.degree["name"] = "bachelor of applied science";
+  const presentationDefinition = getPresentationDefinition();
+  let holderPE: PEJS = new PEJS();
+  const vcs = [...wallet1.verifiableCredentials, ...wallet2.verifiableCredentials];
+  const evaluateResult: EvaluationResults = holderPE.evaluate(presentationDefinition, {
+    "@context": [],
+    type: [],
+    verifiableCredential: vcs,
+    holder: undefined
+  });
+  const selectFromResult: SelectResults = holderPE.selectFrom(presentationDefinition, vcs, [wallet1.holder])
+  const presentationSubmission: PresentationSubmission = holderPE.submissionFrom(presentationDefinition, selectFromResult.verifiableCredentials);
+  console.log("evaluateResult: ", JSON.stringify(evaluateResult, null, 2));
+  console.log("selectFromResult: ", JSON.stringify(selectFromResult, null, 2));
```

Comment:
The result seem to have

 ```
"matches": [
    {
      "rule": "all",
      "matches": [
        "$[2]",
        "$[5]"
      ]
    }
  ],

```

This does not seem correct.

Reply:
that is the correct response. we have two valid credential for one thing
it's up to user to select and send one of them

Reply:
"verifiableCredentials" has only 2 objects these indeces are incorrect. The correct values can only be '0' & '1'
Results file is attached to the ticket.

selectFromResult:  {
  "errors": [
    ...
  ],
  "matches": [
    {
      "rule": "all",
      "matches": [
        "$[2]",
        "$[5]"
      ]
    }
  ],
  "verifiableCredentials": [
    {
      ...
      "credentialSubject": {
        "id": "did:key:z6MkjRagNiMu91DduvCvgEsqLZDVzrJzFrwahc4tXLt9DoHd",
        "degree": {
          "type": "BachelorDegree",
          "name": "Bachelor of Science and Arts"
        }
      },
      ...
    },
    {
      ...
      "credentialSubject": {
        "id": "did:key:z6MkjRagNiMu91DduvCvgEsqLZDVzrJzFrwahc4tXLt9DoHd",
        "degree": {
          "type": "BachelorDegree",
          "name": "bachelor of applied science"
        }
      },
      ...
    }
  ],
  "warnings": []
}

Reply:
Handled in [ESSIFI-158](https://sphereon.atlassian.net/browse/ESSIFI-158)

Reply:
@hrehman-sphereon Do you confirm? If so please resolve